bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/Workspace.java

```diff
@@ -1861,14 +1862,23 @@ protected void initializeMoveDeleteHook() {
 	 * Add the project scope to the preference service's default look-up order so
 	 * people get it for free
 	 */
-	private void initializePreferenceLookupOrder() {
+	private void initializePreferenceLookupOrder() throws CoreException {
 		PreferencesService service = PreferencesService.getDefault();
 		String[] original = service.getDefaultDefaultLookupOrder();
 		List<String> newOrder = new ArrayList<>();
 		// put the project scope first on the list
 		newOrder.add(ProjectScope.SCOPE);
 		newOrder.addAll(Arrays.asList(original));
 		service.setDefaultDefaultLookupOrder(newOrder.toArray(new String[newOrder.size()]));
+		Preferences node = service.getRootNode().node(ProjectScope.SCOPE);
+		if (node instanceof ProjectPreferences) {
+			ProjectPreferences projectPreferences = (ProjectPreferences) node;
+			projectPreferences.setWorkspace(this);
+		} else {
+			throw new CoreException(Status.error(MessageFormat.format(
```

Comment:
Please add **actual** node to the error. Is this `null`? Or wrong type? We don't know here, but we should.

Reply:
done, I think `null` is not a valid value at all the API is required to return always a node and create one if it do not exits.